src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -53,3 +54,86 @@
+         (if (contains? key-uids-set (:key-uid keypair))
+           (update keypair
+                   :accounts
+                   (fn [accounts]
+                     (map (fn [account]
+                            (if (and (contains? key-uids-set (:key-uid account))
+                                     (= (:operable account) ":no"))
+                              (assoc account :operable ":fully")
+                              account))
+                          accounts)))
+           keypair))
+       keypairs))
+
+(rf/reg-event-fx :wallet/make-key-pairs-fully-operable
+ (fn [{:keys [db]} [key-uids-to-update]]
+   (let [key-uids-set (set key-uids-to-update)
+         keypair-name (extract-keypair-name db key-uids-set)]
+     {:db (-> db
+              (update-in [:wallet :accounts] update-accounts key-uids-set)
+              (update-in [:wallet :keypairs] update-keypairs key-uids-set))
+      :fx [[:dispatch
+            [:toasts/upsert
+             {:type  :positive
+              :theme :dark
+              :text  (if (= (count key-uids-to-update) 1)
+                       (i18n/label :t/key-pair-imported-successfully {:name keypair-name})
+                       (i18n/label :t/key-pairs-successfully-imported
+                                   {:count (count key-uids-to-update)}))}]]]})))
+
+(defn- input-connection-string-for-importing-keypairs-keystores-callback
+  [res keypairs-key-uids]
+  (log/info "[local-pairing] input-connection-string-for-importing-keypairs-keystores callback"
+            {:response res
+             :event    :settings/input-connection-string-for-importing-keypairs-keystores-callback})
+  (let [error (when (syncing-events/extract-error res)
+                (str "generic-error: " res))]
+    (when-not (some? error)
+      (rf/dispatch [:wallet/make-key-pairs-fully-operable keypairs-key-uids]))
+    (when (some? error)
+      (rf/dispatch [:toasts/upsert
+                    {:type :negative
+                     :text error}]))))
+
+(defn connection-string-for-key-pair-import
+  [{:keys [db]} [{:keys [sha3-pwd keypairs-key-uids connection-string]}]]
+  (let [key-uid    (get-in db [:profile/profile :key-uid])
+        config-map (.stringify js/JSON
+                               (clj->js
+                                {:receiverConfig
+                                 {:loggedInKeyUid   key-uid
+                                  :keystorePath     ""
+                                  :password         (security/safe-unmask-data
+                                                     sha3-pwd)
+                                  :keypairsToImport keypairs-key-uids}}))]
+    (native-module/input-connection-string-for-importing-keypairs-keystores
+     connection-string
+     config-map
+     #(input-connection-string-for-importing-keypairs-keystores-callback % keypairs-key-uids))))
+
+(rf/reg-event-fx :wallet/connection-string-for-key-pair-import connection-string-for-key-pair-import)
```

Comment:
I think Icaro might have already mentioned this, but I think is a good place to create an Re-Frame effects handler for calling the native-module function. Here's an example of an effects definition: https://github.com/status-im/status-mobile/blob/d2fb6c7ee08857dc988b3914a74479d6e265cf87/src/status_im/contexts/profile/settings/screens/password/change_password/effects.cljs#L8-L17

Reply:
Nice, i was going to ask what's reframe effect and if there is any example. thanks